plugins/inventory/proxmox.py

```diff
@@ -0,0 +1,306 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2016 Guido Günther <agx@sigxcpu.org>, Daniel Lobato Garcia <dlobatog@redhat.com>
+# Copyright (c) 2018 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: proxmox
+    plugin_type: inventory
+    short_description: proxmox inventory source
+    version_added: "2.7"
+    requirements:
+        - requests >= 1.1
+    description:
+        - Get inventory hosts from the proxmox service.
+        - "Uses a configuration file as an inventory source, it must end in ``.proxmox.yml`` or ``.proxmox.yaml`` and has a ``plugin: proxmox`` entry."
+        - Will retrieve the first network interface with an IP for Proxmox nodes
+        - Can retrieve LXC/QEMU configuration as facts
+    extends_documentation_fragment:
+        - inventory_cache
+    options:
+      plugin:
+        description: the name of this plugin, it should alwys be set to 'proxmox' for this plugin to recognize it as it's own.
+        required: True
+        choices: ['proxmox']
```

Comment:
```suggestion
        choices: ['proxmox']
        type: str
```
Please always add types!

(The argument spec isn't yet sanity checked for plugins, so this is only reported in manual review.)

Reply:
Done!